cmd/common.go

```diff
@@ -128,17 +133,34 @@ func setS3OptionsFromCLI(c *cli.Context) *v3.S3BackupConfig {
 	return s3BackupBackend
 }
 
-func doUpgradeLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
-	if _, err := os.Stat(kubeCluster.LocalKubeConfigPath); os.IsNotExist(err) {
-		// there is no kubeconfig. This is a new cluster
-		logrus.Debug("[state] local kubeconfig not found, this is a new cluster")
-		return nil
+func checkLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
+	stateFileExists, err := util.IsFileExists(kubeCluster.StateFilePath)
+	if err != nil {
+		return err
 	}
-	if _, err := os.Stat(kubeCluster.StateFilePath); err == nil {
-		// this cluster has a previous state, I don't need to upgrade!
+	if stateFileExists {
 		logrus.Debug("[state] previous state found, this is not a legacy cluster")
 		return nil
 	}
+	logrus.Debug("[state] previous state not found, possible legacy cluster")
+	return upgradePossibleLegacyCluster(ctx, kubeCluster, fullState, flags)
+}
+
+func upgradePossibleLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
+	kubeConfigExists, err := util.IsFileExists(kubeCluster.LocalKubeConfigPath)
+	if err != nil {
+		return err
+	}
+	if !kubeConfigExists {
+		// if kubeconfig doesn't exist and its a legacy cluster then error out
+		if err := kubeCluster.TunnelHosts(ctx, flags); err != nil {
+			return err
+		}
+		if recoveredCluster := cluster.GetStateFromNodes(ctx, kubeCluster); recoveredCluster != nil {
+			return fmt.Errorf("This is a legacy cluster with no kube config, aborting upgrade")
```

Comment:
Can we put one more line to the error. Suggesting the way to fix it, something like - "Please re-run rke up using legacy rke for the current cluster to fetch and store the kubeconfig"